src/libraries/Microsoft.Extensions.FileProviders.Physical/tests/PhysicalFileProviderTests.netcoreapp.cs

```diff
@@ -0,0 +1,146 @@
+            File.WriteAllText(filePath, "v1.1");
+
+            using var rootOfLink = new DisposableFileSystem();
+            string linkName = Path.GetRandomFileName();
+            string linkPath = Path.Combine(rootOfLink.RootPath, linkName);
+            File.CreateSymbolicLink(linkPath, filePath);
+
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(useWildcard ? "*" : linkName);
+            Assert.False(token.HasChanged);
+
+            // Act
+            File.WriteAllText(filePath, "v1.2");
+            Thread.Sleep(GetTokenPollingInterval(token));
+
+            // Assert
+            Assert.True(token.HasChanged);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void UsePollingFileWatcher_UseActivePolling_HasChanged_SymbolicLink_TargetNotExists(bool useWildcard)
+        {
+            // Arrange
+            using var rootOfLink = new DisposableFileSystem();
+            string linkName = Path.GetRandomFileName();
+            string linkPath = Path.Combine(rootOfLink.RootPath, linkName);
+            File.CreateSymbolicLink(linkPath, "not-existent-file");
+
+            // Act
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(useWildcard ? "*" : linkName);
+
+            // Assert
+            Assert.False(token.HasChanged);
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void UsePollingFileWatcher_UseActivePolling_HasChanged_SymbolicLink_TargetChanged(bool useWildcard)
+        {
+            // Arrange
+            using var rootOfFile = new DisposableFileSystem();
+            string file1Path = Path.Combine(rootOfFile.RootPath, Path.GetRandomFileName());
+            File.WriteAllText(file1Path, "v1.1");
+
+            string file2Path = Path.Combine(rootOfFile.RootPath, Path.GetRandomFileName());
+            File.WriteAllText(file2Path, "v2.1");
+
+            using var rootOfLink = new DisposableFileSystem();
+            string linkName = Path.GetRandomFileName();
+            string linkPath = Path.Combine(rootOfLink.RootPath, linkName);
+            File.CreateSymbolicLink(linkPath, file1Path);
+
+            string filter = useWildcard ? "*" : linkName;
+            using var provider = new PhysicalFileProvider(rootOfLink.RootPath) { UsePollingFileWatcher = true, UseActivePolling = true };
+            IChangeToken token = provider.Watch(filter);
+            Assert.False(token.HasChanged);
+
+            // Act 1 - Change file 1's content.
+            File.WriteAllText(file1Path, "v1.2");
+            Thread.Sleep(GetTokenPollingInterval(token));
+
+            // Assert 1
+            Assert.True(token.HasChanged);
```

Comment:
I think you already test this in UsePollingFileWatcher_UseActivePolling_HasChanged_SymbolicLink, you can leave this out and just make it so that you never change file1Path, only update the target from file1Path to file2Path.